File: src/main/webapp/components/service/driverService.js
(function () {
	'use strict';

	angular.module('hyperonDemoApp').factory('DriverService', DriverService);

	function DriverService($http) {
		var service = {
			getDriver: getDriver,
			updateAddressZipCode: updateAddressZipCode,
			updateAddressCity: updateAddressCity,
			updateAddressStreet: updateAddressStreet,
			updateDriverBirthDate: updateDriverBirthDate,
			updateDriverFirstName: updateDriverFirstName,
			updateDriverGender: updateDriverGender,
			updateDriverLastName: updateDriverLastName,
			updateDriverAccidentCount: updateDriverAccidentCount,
			updateDriverTrafficTicketsCount: updateDriverTrafficTicketsCount,
			updateDriverLicenceObtainedAtAge: updateDriverLicenceObtainedAtAge
		};

		return service;

		function getDriver() {
			return $http.get("/driver");
		}

		function updateAddressZipCode(zipCode) {
			return $http.put("/driver/address/zipCode", zipCode);
		}

		function updateAddressCity(city) {
			return $http.put("/driver/address/city", city);
		}

		function updateAddressStreet(street) {
			return $http.put("/driver/address/street", street);
		}

		function updateDriverBirthDate(birthDate) {
			return $http.put("/driver/birthDate", birthDate);
		}

		function updateDriverFirstName(firstName) {
			return $http.put("/driver/firstName", firstName);
		}

		function updateDriverGender(gender) {
			return $http.put("/driver/gender", gender);
		}

		function updateDriverLastName(lastName) {
			return $http.put("/driver/lastName", lastName);
		}

		function updateDriverAccidentCount(accidentCount) {
			return $http.put("/driver/accidentCount", accidentCount);
		}

		function updateDriverTrafficTicketsCount(trafficTicketsCount) {
			return $http.put("/driver/trafficTicketsCount", trafficTicketsCount);
		}

		function updateDriverLicenceObtainedAtAge(licenceObtainedAtAge) {
			return $http.put("/driver/licenceObtainedAtAge", licenceObtainedAtAge);
		}
	}
})();

File: src/main/java/pl/decerto/higson/demo/motor/service/DriverService.java
package pl.decerto.higson.demo.motor.service;

import java.util.Date;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import pl.decerto.higson.demo.motor.domain.Driver;

@Service
public class DriverService {
	private final QuoteService quoteService;

	@Autowired
	public DriverService(QuoteService quoteService) {
		this.quoteService = quoteService;
	}

	public Driver getDriver() {
		return quoteService.getQuote().getDriver();
	}

	public void updateFirstName(String firstName) {
		getDriver().setFirstName(firstName);
	}

	public void updateGender(String gender) {
		getDriver().setGender(gender);
	}

	public void updateLastName(String lastName) {
		getDriver().setLastName(lastName);
	}

	public void updateBirthDate(Date birthDate) {
		getDriver().setDateOfBirth(birthDate);
	}

	public void updateZipCode(String zipCode) {
		getDriver().getAddress().setZipCode(zipCode);
	}

	public void updateCity(String city) {
		getDriver().getAddress().setCity(city);
	}

	public void updateStreet(String street) {
		getDriver().getAddress().setStreet(street);
	}

	public void updateAccidentCount(Integer accidentCount) {
		getDriver().setNumberOfAccidents(accidentCount);
	}

	public void updateTrafficTicketsCount(Integer trafficTicketsCount) {
		getDriver().setNumberOfTickets(trafficTicketsCount);
	}

	public void licenceObtainedAtAge(Integer licenceObtainedAtAge) {
		getDriver().setLicenceObtainedAtAge(licenceObtainedAtAge);
	}
}
